### ISSUE TITLE
Crash when dealing with out-of-range hook index

### ISSUE THREAD

**Issue by lukespragg** [RustOxide/Oxide-2-Patcher#35]
```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at OxidePatcher.Hooks.Simple.ApplyPatch(MethodDefinition original, ILWeaver weaver, AssemblyDefinition oxideassembly)
   at OxidePatcher.Views.HookViewControl.OnLoad(EventArgs e)
   at System.Windows.Forms.Control.CreateControl(Boolean fIgnoreVisible)
   at System.Windows.Forms.Control.CreateControl(Boolean fIgnoreVisible)
   at System.Windows.Forms.Control.CreateControl()
   at System.Windows.Forms.Control.ControlCollection.Add(Control value)
   at System.Windows.Forms.TabControl.ControlCollection.Add(Control value)
   at OxidePatcher.PatcherForm.AddTab(String name, Control control, Object tag)
   at OxidePatcher.PatcherForm.GotoHook(Hook hook)
   at OxidePatcher.PatcherForm.objectview_NodeMouseDoubleClick(Object sender, TreeNodeMouseClickEventArgs e)
   at System.Windows.Forms.TreeView.OnNodeMouseDoubleClick(TreeNodeMouseClickEventArgs e)
   at System.Windows.Forms.TreeView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**Comment by HellUnit** [comment]
Working on this. Easy enough to get it to pop up a message box to indicate that the specified index is invalid rather than displaying the unhandled exception dialog. But if you try to patch with the invalid index, it will pop up the dialog again, but the text in the PatchProcess form still indicates that the hook was applied. That's the part that's taking me a little longer. I'm thinking of having the ApplyPatch method in Simple.cs return a boolean rather than void to indicate whether or not the patch was successful. It'll take a bit more poking around though as I haven't looked at everything involved.

**Comment by HellUnit** [comment]
Alright, so I've got it where it will display an error dialog and if you attempt to patch with the invalid index, the text in PatchProcessForm will indicate that patching that particular method failed. It also flags the hook in the objectview. Now I'm down to one last problem. Same problem with Flag/Unflag all. Currently open tabs are not updated to reflect the flagged status of the hook. Other than that, I've got it done.
